### PR TITLE
Updated Account Setup and new Account Editing Functionality

### PR DESCRIPTION
**Pull request by Supereg**
# Updated Account Setup and new Account Editing Functionality

## :recycle: Current situation & Problem
There is an upcoming release of SpeziAccount that restructures and simplifies account setup procedure. Further, it adds a new Account Overview that allows you to view and edit your active account details.

While it might not be required for all projects, we currently force a user account (e.g. as the standard expects a user account for all operations) even if the user decides to log out.

Furthermore, we restore test case functionality for some tests cases that broke due to the latest accessibility improvements in some dependencies.

For more information see:
* https://github.com/StanfordSpezi/SpeziAccount/pull/7
* https://github.com/StanfordSpezi/SpeziFirebase/pull/8

## :gear: Release Notes 
* Newly designed Account Setup experience
* New Account Editing Functionality
* Support complete removal of user data

## :books: Documentation
In code documentation was adjusted.

## :white_check_mark: Testing
Tests were updated to cover new vectors.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Supereg** [comment]
This is currently blocked by https://github.com/StanfordSpezi/SpeziOnboarding/issues/21

**Comment by Supereg** [comment]
We still want to incorporate https://github.com/StanfordSpezi/SpeziAccount/pull/24

File: TemplateApplication/Schedule/ScheduleView.swift [hunk 1]
```diff
@@ -46,10 +50,25 @@ struct ScheduleView: View {
...
                 }
+                .toolbar {
+                    if !FeatureFlags.disableFirebase || ProcessInfo.processInfo.isPreviewSimulator {
+                        Button(action: {
+                            presentingAccount = true
+                        }) {
+                            Image(systemName: "person.crop.circle")
+                        }
+                            .accessibilityLabel("ACCOUNT_TITLE")
+                    }
```
Comment (PSchmiedmayer) [inline TemplateApplication/Schedule/ScheduleView.swift:61]: I would suggest to make this button a reusable view so we can also add it to the mock upload tab in the same position 👍

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements @Supereg, great to see the new account module in the Template Application! 🚀

I only had a comment to reuse the account button across the different tabs to display it in a consistent position. Apart from this I am happy to see the PR being merged 👍

File: TemplateApplication/Schedule/ScheduleView.swift (hunk 1 above)
Comment (Supereg) [reply TemplateApplication/Schedule/ScheduleView.swift:61]: I would assume, we also want the button to be present in the contacts view then?

Comment (PSchmiedmayer) [reply TemplateApplication/Schedule/ScheduleView.swift:61]: Oh yes, every tab if possible 👍

Comment (Supereg) [reply TemplateApplication/Schedule/ScheduleView.swift]: Every tab now displays the reusable `AccountButton` view in their navigation bar. 👍

**Pull request merged**